localstack/services/generic_proxy.py

```diff
@@ -402,9 +411,11 @@ def is_full_url(url):
         if isinstance(updated_response, Response):
             response = updated_response
 
-    # allow pre-flight CORS headers by default
-    append_cors_headers(response)
-
+    # Checking whether response is comming from S3 listner
+    from localstack.services.s3.s3_listener import ProxyListenerS3
+    is_s3 = any([True if isinstance(service_listener, ProxyListenerS3) else False for service_listener in listeners])
+    if is_s3:
+        append_cors_headers(response)
```

Comment:
I think I have done this wrong. Logic supposed to be `if not is_s3:`. The intention behind this logic is to prevent the attachment of the CORS headers in the intermediate loops. As each response will be sent from the EdgeProxyListener, we are not able to achieve the logic we discussed ('Put an empty value to CORS headers') because logic is being triggered for all the intermediate loops and it will replace the empty value of the CORS headers. So if we are skipping the `append_cors_headers(response)` method If the listener is from the S3.